website/docs/docs/build/incremental-microbatch.md

```diff
@@ -280,6 +282,123 @@ For now, dbt assumes that all values supplied are in UTC:
+
+### Prerequisites
+
+To enable parallel execution, you must meet the following conditions:
+
+- You use the following supported adapters:
+  - Snowflake
+  - Databricks
+  - More adapters coming soon!
+- You meet [additional conditions](#how-parallel-batch-execution-works) mentioned in the next section
+
+### How parallel batch execution works
+
+A batch can only run in parallel if:
+
+| Step | Condition     |  Parallel execution   | Sequential execution|
+| ---- | ---------------| :------------------: | :----------: |
+| 1.   | **Not** the first batch |  ✅         | -            |
+| 2.   | **Not** the last batch  |  ✅         | -            |
+| 3.   | [Adapter supports](#prerequisites) parallel batches | ✅  | -         |
+
+
+After checking for 1, 2, and 3 in the previous table &mdash; and if `concurrent_batches` value isn't set, dbt will intelligently auto-detect if the model invokes the [`{{ this }}`](/reference/dbt-jinja-functions/this) Jinja function. If it references `{{ this }}`, the batches will run sequentially since  `{{ this }}` represents the database of the current model and referencing the same relation causes conflict. 
+
+Otherwise, if the `concurrent_batches` value isn't set _and_ `{{ this }}` isn't detected (and other conditions are met), the batches will run in parallel.  
+### Parallel or sequential execution?
+
+
+
+Choosing between parallel batch execution and sequential processing depends on the specific requirements of your application. 
+
+- Parallel batch execution is faster but requires logic that's independent of batch execution order. For example, if you're developing a data pipeline for a system that processes user transactions in batches, each batch is executed in parallel for better performance. However, the logic used to process each transaction shouldn't depend on the order of how batches are executed or completed.
+- Sequential processing is slower but essential for calculations like [cumulative metrics](/docs/build/cumulative)  in microbatch models. It processes data in the correct order, allowing each step to build on the previous one.
+
+<!-- You can override the check for `this` by setting `concurrent_batches` to either `True` or `False`. If set to `False`, the batch will be run sequentially. If set to `True` the batch will be run in parallel (assuming [1], [2], and [3])
+To override the `this` check, use the `concurrent_batches` configuration:
+
+
+<File name='dbt_project.yml'>
+
+```yaml
+models:
+  +concurrent_batches: True
+```
+
+</File>
+
+or:
+
+<File name='models/my_model.sql'>
+
+```sql
+{{
+  config(
+    materialized='incremental',
+    concurrent_batches=True,
+    incremental_strategy='microbatch'
+    
+    ...
+  )
+}}
+
+select ...
+```
+
+</File>
+-->
+
+### Configure `concurrent_batches` 
+
+If you meet all the [conditions](#prerequisites), set the `concurrent_batches` config in your `dbt_project.yml` or incremental microbatch model `.sql` file to run batches in parallel:
```

Comment:
@nataliefiann @QMalcolm changed this so it's clear that the config is an override option
```suggestion
By default, dbt auto-detects whether batches can run in parallel for microbatch models, and this works correctly in most cases. However, you can override dbt's detection by setting the `concurrent_batches` config in your `dbt_project.yml` or model `.sql` file to specify parallel or sequential execution, given you meet all the [conditions](#prerequisites):
```